src/comm/Connection.cc

```diff
@@ -172,6 +172,12 @@ Comm::Connection::detailCodeContext(std::ostream &os) const
     return os << Debug::Extra << "connection: " << *this;
 }
 
+bool
+Comm::Connection::closing() const
+{
+    return !isOpen() || fd_table[fd].closing();
```

Comment:
commit 9c33894

> Closed and closing are different things. We should not mislead the caller.

I think a lot before conclude to consider a closed connection as closing. In the view of API consistency you are right. But in most cases actually we need a "closeOrClosing()" method.

Reply:
Thank you for fixing this.

> But in most cases actually we need a "closeOrClosing()" method.

This could be just a communication issue, but FWIW: If you were to say "in _all_ cases, we need a either-closed-or-closing logic, and that is why I implemented that logic, and then, to simplify, I just called the method closing()", then I would understand your logic (and would probably ask you to just rename the method). When you are actually saying "in _most_ cases,...", I cannot understand why you are trying to defend the code that is broken in some cases (i.e. those cases that are not covered by "most").